framework/src/org/checkerframework/common/fieldtypeinference/FieldTypeInferenceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,89 @@
+package org.checkerframework.common.fieldtypeinference;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.util.AnnotatedTypes;
+
+import com.sun.source.tree.CompilationUnitTree;
+
+/**
+ * The ATF of a Checker must be a subtype of this class for it to perform
+ * intra-class inference for private fields.
+ *
+ * @author pbsf
+ *
+ */
+public class FieldTypeInferenceAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+    /** Maps a Receiver representing a private field to an ATM that contains the
+    LUB of all AMs they have been assigned to. **/
```

Comment:
Should "they" be "it"?
Should "they have been assigned to" be "that have been assigned to it"?